src/index.tsx

```diff
@@ -1,33 +1,99 @@
-import React from "react";
-import { type ViewStyle, Text, Platform } from "react-native";
+import React, { useState } from "react";
+import { Platform, type ViewStyle, Text, View, StyleSheet } from "react-native";
 import { Camera, useCameraDevice } from "react-native-vision-camera";
-import { RunningMode, useObjectDetection } from "./objectDetection";
+import {
+  type ResultBundleMap,
+  RunningMode,
+  useObjectDetection,
+} from "./objectDetection";
 
 type MediapipeProps = {
   style: ViewStyle;
+  resultsPanel: boolean;
 };
 
-export const MediapipeCamera: React.FC<MediapipeProps> = ({ style }) => {
-  const device = useCameraDevice("front");
+// TODO:
+//  Keep track of the last non-zero infereance and display that for
+//   some time after the inference is done if the next ones are zero
+//  Make the stats panel scrollable
+//  Clean up the layout
+//  displayName is empty - should this be filled in by the frameProcessor?
+export const MediapipeCamera: React.FC<MediapipeProps> = ({
+  style,
+  resultsPanel,
+}) => {
+  const [stats, setStats] = useState<ResultBundleMap | null>(null);
+  const device = useCameraDevice("back");
   const frameProcessor = useObjectDetection(
     (results) => {
-      console.log(results);
+      setStats(results);
     },
     (error) => {
       console.log(error);
     },
     RunningMode.LIVE_STREAM,
     "efficientdet-lite0.tflite"
   );
-  return device !== undefined ? (
+
+  if (device === undefined) {
+    return <Text>no device</Text>;
+  }
+
+  const camera = (
     <Camera
-      style={style}
+      style={resultsPanel ? styles.box : style}
       device={device}
       pixelFormat={Platform.select({ ios: "rgb", android: "yuv" })}
       isActive={true}
       frameProcessor={frameProcessor}
     />
+  );
+
+  let rows: React.JSX.Element[] = [];
+  let count = 0;
+
+  if ((stats?.results.length ?? 0) > 1) {
+    console.log("more than one result", stats?.results.length);
+  }
+
+  if (stats?.results && stats.results.length > 0 && stats.results[0]) {
+    count = stats.results[0].detections.length;
+    rows = stats.results[0].detections.map((obj) => {
+      const categories = obj.categories
+        .map((cat) => `${cat.categoryName} (${cat.score})`)
+        .join(", ");
+      const text = `${categories}: ${JSON.stringify(obj.boundingBox)}`;
+      console.log(text);
+      return <Text>{text}</Text>;
+    });
+  }
```

Comment:
While this code is technically correct, it does things in a very un-react-y way. 

1. prefer const
2. prefer to do rendering inline in the markup
3. leverage javascript's conciseness
4. use hooks (in this case `React.useMemo`) to eliminate needless computation on every render

```typescript
const rows = React.useMemo(()=> (stats?.results ?? []).map( r => { 
    const categories = r.categories.map(...).join(", ");
    return `${categories}: ${JSON.stringify(obj.boundingBox)}`;
}), [stats?.results]);

return (
    ....
    <Text>count : {rows.count}</Text>
    {rows.map( r => (<Text>{r}</Text>))}
   ...
);
```

Reply:
Thanks! There will probably be a decent amount of this as I ramp up since Vue and React are similar but different in ways that will likely cause things like this.